idf_build_apps/junit/report.py

```diff
@@ -0,0 +1,164 @@
+# SPDX-FileCopyrightText: 2023 Espressif Systems (Shanghai) CO LTD
+# SPDX-License-Identifier: Apache-2.0
+
+"""
+The test report should look like something like this:
+
+<testsuites>
+    <testsuite
+            errors="0"
+            failures="1"
+            skipped="0"
+            tests="29"
+            name="Build target esp32 in example folder"
+            hostname="GARM-C2-WX-1-R5S4N264"
+            ... # more machine specs
+            time="1051.215"
+            timestamp="2022-12-15T13:25:17.689181">
+        <testcase
+                file="examples/get_started/hello_world"
+                name="esp32.default.examples/get-started/hello_world"
+                time="60.2454">
+            <failure message="build warnings or error message">
```

Comment:
Seems like `system-out` and `system-err` elements are typically used to capture stdout and stderr, respectively.
(e.g. https://github.com/kyrus/python-junit-xml and https://github.com/junit-team/junit5/commit/e89948d0efdaa55a5c88fe7f0be47cd2b3788e43)

Reply:
@igrr Here I'm following the gitlab junit format.

- gitlab doc: https://docs.gitlab.com/ee/ci/testing/unit_test_reports.html#how-it-works
- reference it linked from IBM: https://www.ibm.com/docs/en/developer-for-zos/16.0?topic=formats-junit-xml-format